packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -259,6 +265,55 @@ export class PanelPlugin<
     return this;
   }
 
+  /**
+   * Tells Grafana if the plugin should subscribe to annotation results.
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *     .useFieldConfig({})
+   *     ...
+   *     ...
+   *     .setAnnotationSupport(true);
+   * ```
+   *
+   * @public
+   **/
+  setAnnotationSupport(supported: boolean) {
+    this.supportsAnnotations = supported;
+    return this;
+  }
+
+  /**
+   * @deprecated setAlertStatesSupport is deprecated and will be removed when the next generation alerting is in place
+   * Tells Grafana if the plugin should subscribe to alert state results.
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *     .useFieldConfig({})
+   *     ...
+   *     ...
+   *     .setAlertStatesSupport(true);
+   * ```
+   *
+   * @public
+   **/
+  setAlertStateSupport(supported: boolean) {
+    this.supportsAlertStates = supported;
+    return this;
```

Comment:
I think this can be combined like you had it before, with an object with options (with the alertState boolean, and annotations)